api/provision/consume-provisioning-job.ts

```diff
@@ -0,0 +1,65 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { ProvisionRequest } from "../../models/provisioning-jobs";
+import { sqsClient } from "../../utils/aws-sdk/sqs";
+import middy from "@middy/core";
+import { DeleteMessageCommand, ReceiveMessageCommand, ReceiveMessageCommandInput } from "@aws-sdk/client-sqs";
+import { ApiSuccessResponse, SuccessStatusCode } from "../../utils/response";
+import { IpCheckerMiddleware } from "../../utils/middleware/ip-logging";
+import { errorHandlingMiddleware } from "../../utils/middleware/error-handling-middleware";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+
+const PROVISIONING_QUEUE_URL = process.env.PROVISIONING_QUEUE_URL ?? "";
+const logger = new Logger({ serviceName: "provisioningWorkflow" });
+
+export async function baseHandler(event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> {
+  logger.info("General lambda info", { isAColdStart: logger.isColdStart() });
+  // poll messages from the queue
+  const receiveMessageInput: ReceiveMessageCommandInput = {
+    QueueUrl: PROVISIONING_QUEUE_URL,
+    MaxNumberOfMessages: 10,
```

Comment:
Has there been discussion of parameterizing this quantity?  This is good for now.

Reply:
No I don't think so.